xlsx/src/import/worksheets/mod.rs

```diff
@@ -0,0 +1,535 @@
+    util::{get_attribute, get_color},
+};
+
+mod worksheet_streaming;
+mod worksheet_tree;
+
+pub(crate) struct Sheet {
+    pub(crate) name: String,
+    pub(crate) sheet_id: u32,
+    pub(crate) id: String,
+    pub(crate) state: SheetState,
+}
+
+pub(crate) struct WorkbookXML {
+    pub(crate) worksheets: Vec<Sheet>,
+    pub(crate) defined_names: Vec<DefinedName>,
+}
+
+pub(crate) struct Relationship {
+    pub(crate) target: String,
+    pub(crate) rel_type: String,
+}
+
+pub(super) struct SheetSettings {
+    pub id: u32,
+    pub name: String,
+    pub state: SheetState,
+    pub comments: Vec<Comment>,
+}
+
+#[derive(Debug, Clone)]
+struct SheetView {
+    is_selected: bool,
+    selected_row: i32,
+    selected_column: i32,
+    frozen_columns: i32,
+    frozen_rows: i32,
+    range: [i32; 4],
+    show_grid_lines: bool,
+}
+
+impl Default for SheetView {
+    fn default() -> Self {
+        Self {
+            is_selected: false,
+            selected_row: 1,
+            selected_column: 1,
+            frozen_rows: 0,
+            frozen_columns: 0,
+            range: [1, 1, 1, 1],
+            show_grid_lines: true,
+        }
+    }
+}
+
+pub(super) fn load_sheets<R: Read + std::io::Seek>(
+    archive: &mut zip::read::ZipArchive<R>,
+    rels: &HashMap<String, Relationship>,
+    workbook: &WorkbookXML,
+    tables: &mut HashMap<String, Table>,
+    shared_strings: &mut Vec<String>,
+) -> Result<(Vec<Worksheet>, u32), XlsxError> {
+    // load comments and tables
+    let mut comments = HashMap::new();
+    for sheet in &workbook.worksheets {
+        let rel = &rels[&sheet.id];
+        if rel.rel_type.ends_with("worksheet") {
+            let path = &rel.target;
+            let path = if let Some(p) = path.strip_prefix('/') {
+                p.to_string()
+            } else {
+                format!("xl/{path}")
+            };
+            comments.insert(
+                &sheet.id,
+                load_sheet_rels(archive, &path, tables, &sheet.name)?,
+            );
+        }
+    }
+
+    // load all sheets
+    let worksheets: &Vec<String> = &workbook.worksheets.iter().map(|s| s.name.clone()).collect();
+    let mut sheets = Vec::new();
+    let mut selected_sheet = 0;
+    let mut sheet_index = 0;
+    for sheet in &workbook.worksheets {
+        let sheet_name = &sheet.name;
+        let rel_id = &sheet.id;
+        let state = &sheet.state;
+        let rel = &rels[rel_id];
+        if rel.rel_type.ends_with("worksheet") {
+            let path = &rel.target;
+            let path = if let Some(p) = path.strip_prefix('/') {
+                p.to_string()
+            } else {
+                format!("xl/{path}")
+            };
+            let settings = SheetSettings {
+                name: sheet_name.to_string(),
+                id: sheet.sheet_id,
+                state: state.clone(),
+                comments: comments.get(rel_id).expect("").to_vec(),
+            };
+
+            let streaming = true;
```

Comment:
Here's where we can switch between the tree and streaming parsers. Same questions apply to this variable as did to the variable by the same name in shared_strings.rs.

Reply:
Removed in latest PR